Assets/BowAndArrowImproved/Scripts/HealthBar.cs

```diff
@@ -27,9 +28,12 @@ public void OnTookHit()
         Destroy(lastSquare);
     }
 
-    private void SetupHealthBar()
+    public void SetupHealthBar()
     {
-        for (int i = 0; i < hittable.health; i++)
+        hittable.SetCount(health);

```

Comment:
UI classes DO NOT set things like health, they only consume it to show the UI, not good